msticpy/datamodel/entities/soc/incident.py

```diff
@@ -0,0 +1,169 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Incident Entity class."""
+from datetime import datetime
+from typing import Any, List, Mapping, Optional, Dict
+import pandas as pd
+
+from ...._version import VERSION
+from ....common.utility import export
+from ...entities.entity import Entity, camelcase_property_names
+from ...entities.alert import Alert
+
+__version__ = VERSION
+__author__ = "Pete Bryan"
+
+
+# pylint: disable=invalid-name, too-many-instance-attributes
+@export
+class Incident(Entity):
```

Comment:
Should this be an entity?
I guess technically not but maybe it helps with the edges and stuff.
One thing that we should do for this and the Alert class is - whenever we add an entity or alert to part of the parents collection (e.g. in this case alerts) we should add an edge to the alert.
That would also make the graph generation easier.
Same thing with alerts and it entities member.